Add code coverage and cover more code

jason-o-matic opened the pull request.

jason-o-matic committed:
Add code coverage

jason-o-matic committed:
Add a test for rails middleware integration

jason-o-matic committed:
Add basic tests for Honeybadger and fix issue with double counting

This code wasn't covered at all before.

jason-o-matic committed:
Track coverage across multiple rubies and gemfiles

Otherwise one of dalli or memcached won't get tracked in our current
setup.

jason-o-matic committed:
Add tests for sidekiq per job tracing

janxious commented on spec/support/test_rails_app.rb:
```diff
@@ -0,0 +1,21 @@
+require "rails/all"
```
what is this app for?

jason-o-matic replied:
We didn't test the middleware injection in rails previously, but now this does.

janxious replied:
gotcha

janxious commented on spec/metrician_spec.rb:
```diff
@@ -60,6 +60,51 @@ def logger; end
     end
   end
 
+  describe "exception tracking" do
```
good catch.

janxious commented on spec/metrician_spec.rb:
```diff
@@ -412,5 +412,25 @@ def app
+
+    describe "rails" do
+      def app
+        Rails.application
```
Re: https://github.com/Instrumental/metrician-ruby/pull/28/commits/b090ae7712ac0e6d630c1cf3347f0d09ded197e9#r131264357, I think it would be clearer if we could refer to this application by name `TestRailsApp`

janxious reviewed (approved):
one thought, but 👍

jason-o-matic committed:
Clarified Rails instance reference

Per PR feedback, this more clearly indicates where you might look for
more info about the rails app.

It's not entirely clear to me how stable this API is, so if there are
problems later we can always fall back to `Rails.application`.

jason-o-matic committed:
Make code highlighting work slightly better

jason-o-matic committed:
Hook into both memcached and dalli if both are available

While working on code coverage I discovered that we only pick one when
we could hook into both if available. This also smooths out the tests
a bit related to code coverage because different runs cover the same
code paths.
